toppra/dracula/__init__.py

```diff
@@ -146,6 +147,35 @@ def lims_obeyed(self, traj, raise_2nd_order):
                 return False
         return True
 
+    def joint_limits_obeyed(self, traj, multiplier):
+        lims_obeyed = True
+        t_max = 1 if multiplier is None else multiplier * self.t_sum
+        t_sample = np.arange(0, t_max, 0.001)
+        q_sample = traj(t_sample)
+
+        t_index = 0
+        waypt_copy = [self.waypts[0]]
```

Comment:
`waypt_copy` will be of the same shape as `waypts`, right? if shape is known, init with zeros or `waypts.copy()` to preallocate memory and modify its slices

Reply:
No it won't.  It is at least as big as waypts but the amount bigger depends on the number of segments with joint limit violations.